### ISSUE TITLE
Extract refactor can produce `unnecessary_parenthesis` trigger

### ISSUE THREAD

**Issue by FMorschel** [dart-lang/sdk#56944]
Repro:

```dart
class A {
  void a() {
    (1 + 2);
  }
}
```

Now select `(1 + 2)` and the option to `Extract Method`.

Output:

```dart
class A {
  void a() {
    b();
  }

  void b() => (1 + 2);   // <-- unnecessary_parenthesis
}
```

**Comment by bwilkerson** [comment]
It's true that there are unnecessary parentheses in the extracted method, but there were unnecessary parentheses in the code before it was extracted. The fact that it wasn't reported is a bug in the lint (I opened https://github.com/dart-lang/linter/issues/5112 to report it).

IMO, the extract method refactoring should not change the code being extracted. If there are diagnostics being reported against it, they should remain and users can fix the diagnostics separately.

**Comment by FMorschel** [comment]
I agree, but when it is a simple expression it could use the `unparenthesized` version. Say:

```dart
void foo() => (bar() + baz()) ? 0 : 1;
```

Even with https://github.com/dart-lang/linter/issues/4996 merged, there would be no lint on `(bar() + baz())` but it would trigger on the extracted code.

Sorry for the bad first example. This one is more similar to the actual code I found the behaviour.

**Comment by bwilkerson** [comment]
Understood. And I'm not saying that the refactor can't introduce a lint violation. I just think it's better if the refactoring doesn't try to fix the code. But clearly not everyone agrees with me 😄 , and that's fine.
